frontend/src/components/Divisions.tsx

```diff
@@ -145,24 +136,73 @@ const Divisions = ({
     };
   };
 
-  const addDivision = (inputData: DivisionFormTypes) => {
-    console.log(inputData);
-    onRefetch();
-    setIsAddModalOpen(false);
+  // simple actions
+  const onAdd = () => {
+    setIsAddModalOpen(true);
+  };
+
+  const onView = (row: IDivision) => {
+    alert(`Not yet implemented, ${row}`);
+  };
+
+  const onEdit = (row: IDivision) => {
+    setSelectedIdForEdit(row.id);
   };
 
-  const editDivision = (inputData: DivisionFormTypes) => {
-    console.log(inputData);
-    onRefetch();
-    setIsEditModalOpen(false);
+  const onDelete = (row: IDivision) => {
+    alert(`Not yet implemented, ${row}`);
   };
 
-  const closeAddModal = () => {
-    setIsAddModalOpen(false);
+  // add division
+  const handleAdd = (inputData: DivisionFormTypes) => {
+    addDivisionMutation(inputData.name, {
+      onSuccess: () => {
+        useSuccessToast(i18n.t('division:success.add'));
+        onRefetch();
+      },
+      onError: () => {
+        useErrorToast(
+          i18n.t('division:errors.generic.add', {
+            division: i18n.t(`division:errors.${divisionType}`),
+          }),
+        );
+      },
+      onSettled: () => {
+        setIsAddModalOpen(false);
+      },
+    });
   };
 
-  const closeEditModal = () => {
-    setIsEditModalOpen(false);
+  // edit division
+  const handleEdit = (inputData: DivisionFormTypes) => {
+    if (selectedIdForEdit) {
+      editDivisionMutation(
+        { id: selectedIdForEdit, name: inputData.name },
+        {
+          onSuccess: () => {
+            useSuccessToast(i18n.t('division:success.edit'));
+            onRefetch();
+          },
+          onError: () => {
+            if (editDivisionMutationError)
```

Comment:
this error comes on onError callback, we don't need to get it from the hook, since we use the callback we should get it from there.